### ISSUE TITLE
Support for importing relation objects by ID from json file

### ISSUE THREAD

**Issue by omidraha** [realm/realm-java#4627]
Hi,

The `createOrUpdateAllFromJson` method or similar other methods, 
support for importing objects by putting only `id` for relation objects, 
instead of putting whole object data on the json file.

For example when we have these two  `Dog` and  `Person` models:

```java
public class Dog extends RealmObject {
    @PrimaryKey
    private int id;
    private String name;

}

public class Person extends RealmObject {
    @PrimaryKey
    private int id;
    private String name;
    private RealmList<Dog> dogs;

}
```

And assumed we currently have these records on the `Dog` model on the database:

![table-diagram](https://cloud.githubusercontent.com/assets/397672/25946665/6d7ffbfa-3661-11e7-8b22-435453fd6af6.png)

Now, we want to able to import `Person` model from this `Person.json` file:

```[{"id":1, "name": "Jane", "dogs": [1,2]]```

Instead of this `Person.json` file:

```[{"id":1, "name": "Jane", "dogs": [{"id": 1, "name": "Fido"}, {"id": 2, "name": "Fluffy"}]```

by doing:

```java
InputStream stream = getAssets().open("Person.json");
Realm realm = Realm.getDefaultInstance();
realm.beginTransaction();
realm.createOrUpdateAllFromJson(Person.class, stream);
realm.commitTransaction();
```

**Comment by cmelchior** [comment]
Yes, a similar idea has been raised here as well: https://github.com/realm/realm-java/pull/3758. Personally, I find the idea compelling and would definitely ease integration with a lot of REST API's that just send ID's.

**Comment by hardysim** [comment]
I've just created #6666 before finding this but it seems we're having the same problem. Maybe you can use the workaround mentioned there until realm adds this functionality.
